pennylane/operation.py

```diff
@@ -900,6 +910,48 @@ def matrix(self):
         # over the defined wires.
         return functools.reduce(np.kron, U_list)
 
+    def prune(self):
+        """Returns a pruned tensor product of observables by removing :class:`~.Identity` instances from
+        the observables building up the :class:`~.Tensor`.
+
+        The return_type attribute is preserved while pruning.
+
+        If the tensor product only contains one observable, then this observable instance is
+        returned.
+
+        Note that this way this method might return an instance that is not a :class:`~.Tensor`
+        instance.
+
+        **Example:**
+
+        1.
+
+        >>> O = qml.PauliZ(0) @ qml.Identity(1) @ qml.PauliZ(2)
+        >>> pruned_obs = O.prune()
+        <pennylane.operation.Tensor at 0x7fc1642d1590
+        >>> pruned_obs.obs
+        [<pennylane.ops.qubit.PauliZ object at 0x7fc1b076afd0>, <pennylane.ops.qubit.PauliZ object at 0x7fc1b075bd90>]
+
+        2.
+
+        >>> O = qml.PauliZ(0) @ qml.Identity(1)
+        >>> O.prune()
+        <pennylane.ops.qubit.PauliZ at 0x7fc1642d1850>
+
+        Returns:
+            ~.Observable: the pruned tensor product of observables
+        """
+        # Return one Identity in there in case the tensor only contains Identities
```

Comment:
I would suggest moving this comment underneath the first if statement so that it is clearer which line it applies to:
```python
if len(self.non_identity_obs) == 0:
    # tensor only contains identity observables, so simply return the identity
    obs = qml.Identity(0)
```